dockstore-webservice/src/main/java/io/dockstore/webservice/core/Tool.java

```diff
@@ -93,6 +93,7 @@
         @NamedQuery(name = "io.dockstore.webservice.core.Tool.findByUserRegistryNamespace", query = "SELECT t from Tool t WHERE t.id in (SELECT ue.id FROM User u INNER JOIN u.entries ue where u.id = :userId) AND t.registry = :registry AND t.namespace = :namespace"),
         @NamedQuery(name = "io.dockstore.webservice.core.Tool.findByUserRegistryNamespaceRepository", query = "SELECT t from Tool t WHERE t.id in (SELECT ue.id FROM User u INNER JOIN u.entries ue where u.id = :userId) AND t.registry = :registry AND t.namespace = :namespace AND t.name = :repository"),
         @NamedQuery(name = "io.dockstore.webservice.core.Tool.getEntriesByUserId", query = "SELECT t FROM Tool t WHERE t.id in (SELECT ue.id FROM User u INNER JOIN u.entries ue where u.id = :userId)"),
+        @NamedQuery(name = "io.dockstore.webservice.core.Tool.getPublishedNamespaces", query = "SELECT lower(namespace) FROM Tool c WHERE c.isPublished = true"),
```

Comment:
Shouldn't the query specify `distinct`?

Reply:
Ah, I see further below you put the result into a Java `Set`, so they become distinct then. Still, might as well throw the work on the DB, even though we're talking small amounts of data here.